lang/rustup/rustc-triple.mk

```diff
@@ -0,0 +1,76 @@
+# To enable rustc/cargo support in your packages, add:
+# include $(TOPDIR)/feeds/packages/lang/rustup/rustc-triple.mk
+# below include $(TOPDIR)/rules.mk
+#
+# You will need to add --host=$(RUSTC_TARGET_ARCH) to your packages
+# CONFIGURE_VARS to cross-compile for the target architecture
+
+RUSTC_ARCH_TARGETS:= \
+aarch64-unknown-linux-gnu aarch64-unknown-linux-musl \
+\
+arm-unknown-linux-gnueabi arm-unknown-linux-gnueabihf \
+arm-unknown-linux-musleabi arm-unknown-linux-musleabihf \
+\
+armv4t-unknown-linux-gnueabi \
+\
+armv5te-unknown-linux-gnueabi armv5te-unknown-linux-musleabi \
+\
+armv7-unknown-linux-gnueabi armv7-unknown-linux-gnueabihf \
+armv7-unknown-linux-musleabi \
+\
+hexagon-unknown-linux-musl \
+\
+i586-unknown-linux-gnu i586-unknown-linux-musl \
+\
+i686-unknown-linux-gnu i686-unknown-linux-musl \
+\
+mips-unknown-linux-gnu mips-unknown-linux-musl mips-unknown-linux-uclibc \
+\
+mips64-unknown-linux-gnuabi64 mips64-unknown-linux-muslabi64 \
+\
+mips64el-unknown-linux-gnuabi64 mips64el-unknown-linux-muslabi64 \
+\
+mipsel-unknown-linux-gnu mipsel-unknown-linux-musl \
+mipsel-unknown-linux-uclibc \
+\
+mipsisa32r6-unknown-linux-gnu mipsisa32r6el-unknown-linux-gnu \
+\
+mipsisa64r6-unknown-linux-gnuabi64 mipsisa64r6el-unknown-linux-gnuabi64 \
+\
+powerpc-unknown-linux-gnu powerpc-unknown-linux-musl \
+\
+powerpc64-unknown-linux-gnu powerpc64-unknown-linux-musl \
+\
+powerpc64le-unknown-linux-gnu powerpc64le-unknown-linux-musl \
+\
+riscv64gc-unknown-linux-gnu \
+\
+s390x-unknown-linux-gnu \
+\
+sparc-unknown-linux-gnu \
+\
+sparc64-unknown-linux-gnu \
+\
+thumbv7neon-unknown-linux-gnueabihf thumbv7neon-unknown-linux-musleabihf \
+\
+x86_64-unknown-linux-gnu x86_64-unknown-linux-musl
```

Comment:
Where does this list come from, and I wonder if there is a more maintainable way of organizing this list, e.g. `unknown-linux` is present in every item.

Reply:
Also, it seems that this list of targets is used for two purposes, to test if the target platform is supported, and to have a target to pass to rustc.

My previous comment (organizing the list in a more maintainable way) goes towards testing if the target platform is supported.

To have a target to pass to rustc, I think it would be nice if [`GNU_TARGET_NAME` or `REAL_GNU_TARGET_NAME`](https://github.com/openwrt/openwrt/blob/bcf0704bd2ca852b1cbfa481095afb73e78287f9/rules.mk#L125-L126) can be used instead (although I have no idea if rustc will accept it or not).

Reply:
This list comes from ./rustup.sh --list-targets and has been trimmed to only include MUSL, Glibc and uclibc targets.

At least in the Suricata example, the rustc "triple" (totally misnamed these days) has to be passed in the CONFIGURE_ARGS as a --target=$(RUSTC_TARGET_ARCH) so it knows it's cross-compiling.  The triple is based on the above list.  So, yes, it does do dual duty of not only seeing what the target arch is based on the CONFIG_ flags, but then passes that onto whatever is going to use rustc/cargo to build with.  

Originally, i was looking for a way to just export RUSTC_TARGET_ARCH to the build system so I could pull it, but since that didn't work, I ended up going with the rustc-triple.mk file.

Reply:
GNU_TARGET_NAME/GNU_HOST_NAME triples do not seem to be compat with what the rust triples are.